src/ucp/wireup/select.c

```diff
@@ -1006,23 +1006,24 @@ static double ucp_wireup_aux_score_func(const ucp_worker_iface_t *wiface,
 }
 
 static void ucp_wireup_fill_aux_criteria(ucp_wireup_criteria_t *criteria,
-                                         unsigned ep_init_flags)
+                                         unsigned ep_init_flags,
+                                         unsigned mandatory_flags)
 {
     criteria->title          = "auxiliary";
     criteria->local_md_flags = 0;
     ucp_wireup_init_select_flags(&criteria->local_iface_flags,
                                  UCT_IFACE_FLAG_AM_BCOPY |
                                  UCT_IFACE_FLAG_PENDING, 0);
     ucp_wireup_init_select_flags(&criteria->remote_iface_flags,
-                                 UCP_ADDR_IFACE_FLAG_AM_SYNC, 0);
+                                 UCP_ADDR_IFACE_FLAG_AM_SYNC | mandatory_flags,
```

Comment:
but now we add UCP_ADDR_IFACE_FLAG_CB_ASYNC even when has_cm (line 1022) is true

Reply:
Should it let ```ucp_wireup_fill_aux_criteria``` check whether it's CM lane to add the mandatory_flags or not?

In this PR, it assumes that the caller decides whether it's needs to add extra ```mandatory_flags```.
Until now, we still keep the existing logic:
1. [ucp_wireup_select_wireup_msg_lane](https://github.com/openucx/ucx/blob/v1.15.0-rc1/src/ucp/wireup/select.c#L1883)
   The ```ep_init_flags``` indicate it could be CM lane, so this PR set ```mandatory_flags=0``` here.
2. In [ucp_wireup_select_aux_transport](https://github.com/openucx/ucx/pull/9061/files#diff-ff110fb0f2e163038a3eb15545d3b81d9d32c121b79e7a764b72fbb1fa643470R2372-R2389), it set ```mandatory_flags = UCP_ADDR_IFACE_FLAG_CB_ASYNC```, then do fallback if there's such existing tl.

Reply:
Got it. I'll change the code.